lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
I mentioned it in this [comment, third line](https://github.com/metarhia/jstp/pull/254#discussion_r127193334). It'll grow (be replaced with a bigger one) when message that has arrived cannot fit to the current one.